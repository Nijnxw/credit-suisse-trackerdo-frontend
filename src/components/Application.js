import { Toaster } from "react-hot-toast";
import { BrowserRouter as Router, Redirect, Route, Switch } from "react-router-dom";
import Dashboard from "./dashboard";

function Application() {
  const redirectToDashboard = () => <Redirect to="/app" />
  return (
    <Router>
      <div className="Application">
        <div><Toaster position="bottom-right"/></div>
        <div className="Application__main">
          <Switch>
            <Route path='/app' component={toDashboard} />
            <Route exact path='/' render={redirectToDashboard} />
          </Switch>
        </div>
      </div>
    </Router>
  );
}

const toDashboard = () => <Dashboard />

export default Application;
