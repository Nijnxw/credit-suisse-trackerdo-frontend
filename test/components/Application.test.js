import React from 'react'
import { shallow } from 'enzyme'
import Application from "../../src/components/Application"

let wrapper

beforeEach(() => {
  wrapper = shallow(<Application /> );
})

describe('<Application /> rendering', () => {
  it('should render correctly', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  })
  
  it('should render 1 <div>', () => {
    expect(wrapper.children('div')).toHaveLength(1);
  })
})
