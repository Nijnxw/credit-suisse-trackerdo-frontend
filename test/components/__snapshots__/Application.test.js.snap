// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Application /> rendering should render correctly 1`] = `
"<BrowserRouter>
  <div className=\\"Application\\">
    <div>
      <Toaster position=\\"bottom-right\\" />
    </div>
    <div className=\\"Application__main\\">
      <Switch>
        <Route path=\\"/app\\" component={[Function: toDashboard]} />
        <Route exact={true} path=\\"/\\" render={[Function: redirectToDashboard]} />
      </Switch>
    </div>
  </div>
</BrowserRouter>"
`;
